libraries/cyclestreets-view/src/main/java/net/cyclestreets/views/overlay/PhotoMarkers.kt

```diff
@@ -18,36 +19,29 @@ private const val NEUTRAL = "neutral"
 
 private const val GENERAL = "general"
 
-class PhotoMarkers(private val res: Resources) {
-    private val defaultMarkers: Map<String, Drawable>
-    private val markers = HashMap<String, Drawable>()
-    private val bfo: BitmapFactory.Options
 
-    init {
-        defaultMarkers = mapOf(
+class PhotoMarkers(val res: Resources) {
+    private val markers = HashMap<String, Drawable>()
+    private val defaultMarkers = mapOf(
             GOOD to getMarker(GENERAL, GOOD),
             BAD to getMarker(GENERAL, BAD),
             NEUTRAL to getMarker(GENERAL, NEUTRAL)
-        )
-        bfo = BitmapFactory.Options()
-        bfo.inTargetDensity = 240
-    }
+    )
 
     fun getMarker(photo: Photo): Drawable {
         return getMarker(photo.category(), mapMetaCat(photo.metacategory()))
     }
 
     private fun getMarker(category: String, metaCategory: String): Drawable {
-        val key = String.format("photomarkers/%s_%s.png", category, metaCategory)
+
+        val key = "pm_${category}_${metaCategory}"
 
         if (!markers.containsKey(key)) {
             try {
-                val asset = res.assets.open(key)
-                val bmp = BitmapFactory.decodeStream(asset)
-                val marker = BitmapDrawable(res, bmp)
-                asset.close()
+                val resId = res.getIdentifier(key, "drawable", res.getResourcePackageName(R.drawable.pm_bicycles_bad))
```

Comment:
Can we hoist the res.getResourcePackageName call out of here? Make it a val member perhaps?

Reply:
Otherwise, this is all lovely.

Reply:
Done & done.  Good call.